src/main/java/bankequipmentstatfilter/BankEquipmentStatFilterPanel.java

```diff
@@ -79,20 +103,66 @@ public class BankEquipmentStatFilterPanel extends PluginPanel
         itemsPanel = new JPanel();
         itemsPanel.setBackground(ColorScheme.DARK_GRAY_COLOR);
         itemsPanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+        itemsPanel.setLayout(new BoxLayout(itemsPanel, BoxLayout.Y_AXIS));
 
-        add(itemsPanel, BorderLayout.SOUTH);
+        add(itemsPanel);
     }
 
-    public void displayItems(ItemWithStat[] items, EquipmentStat statType)
+    public void displayItems(Map<Integer, List<ItemWithStat>> items, EquipmentStat statType)
     {
+
         itemsPanel.removeAll();
 
-        final int rowCorrection = items.length % COLUMN_SIZE > 0 ? 1 : 0;
-        final int rowSize = items.length/COLUMN_SIZE + rowCorrection;
-        itemsPanel.setLayout(new GridLayout(rowSize, COLUMN_SIZE, 1, 1));
+        int itemCount = 0;
+        // loop through items
+        for(Map.Entry<Integer, List<ItemWithStat>> entry : items.entrySet()) {
+            Integer slotIdx = entry.getKey();
+            List<ItemWithStat> slotItems = entry.getValue()
+                    .stream().limit(config.maxItemsPerSlot())
```

Comment:
@jkryspin one small change, lets only apply this limit when allSlots is true. This limit is good for all slots, but I think when it's just a single slot it makes more sense to show all the items. Otherwise, this looks great, thanks!